service/gcs/transport/mock.go

```diff
@@ -15,43 +19,54 @@ type MockTransport struct {
 
 // Dial ignores the port, and returns a MockTransport struct.
 func (t *MockTransport) Dial(_ uint32) (Connection, error) {
-	csr, csw := io.Pipe()
-	scr, scw := io.Pipe()
+	sockDir, err := ioutil.TempDir("", "mocktport")
```

Comment:
Why can we do here with unix sockets we couldnt with pipes? I am confused why we need a "more full" moc impl